cmd/lora/main.go

```diff
@@ -0,0 +1,225 @@
+		kitprometheus.NewCounterFrom(stdprometheus.CounterOpts{
+			Namespace: "lora_adapter",
+			Subsystem: "api",
+			Name:      "request_count",
+			Help:      "Number of requests received.",
+		}, []string{"method"}),
+		kitprometheus.NewSummaryFrom(stdprometheus.SummaryOpts{
+			Namespace: "lora_adapter",
+			Subsystem: "api",
+			Name:      "request_latency_microseconds",
+			Help:      "Total duration of requests in microseconds.",
+		}, []string{"method"}),
+	)
+
+	go subscribeToLoRaBroker(svc, mqttConn, natsConn, logger)
+	go subscribeToThingsES(svc, esConn, cfg.instanceName, logger)
+
+	errs := make(chan error, 1)
+
+	go func() {
+		c := make(chan os.Signal)
+		signal.Notify(c, syscall.SIGINT)
+		errs <- fmt.Errorf("%s", <-c)
+	}()
+
+	err = <-errs
+	logger.Error(fmt.Sprintf("LoRa adapter terminated: %s", err))
+}
+
+func loadConfig() config {
+	return config{
+		httpPort:     mainflux.Env(envHTTPPort, defHTTPPort),
+		loraMsgURL:   mainflux.Env(envLoraMsgURL, defLoraMsgURL),
+		natsURL:      mainflux.Env(envNatsURL, defNatsURL),
+		logLevel:     mainflux.Env(envLogLevel, defLogLevel),
+		esURL:        mainflux.Env(envESURL, defESURL),
+		esPass:       mainflux.Env(envESPass, defESPass),
+		esDB:         mainflux.Env(envESDB, defESDB),
+		instanceName: mainflux.Env(envInstanceName, defInstanceName),
+		routeMapURL:  mainflux.Env(envRouteMapURL, defRouteMapURL),
+		routeMapPass: mainflux.Env(envRouteMapPass, defRouteMapPass),
+		routeMapDB:   mainflux.Env(envRouteMapDB, defRouteMapDB),
+	}
+}
+
+func connectToNATS(url string, logger logger.Logger) *nats.Conn {
+	conn, err := nats.Connect(url)
+	if err != nil {
+		logger.Error(fmt.Sprintf("Failed to connect to NATS: %s", err))
+		os.Exit(1)
+	}
+
+	return conn
+}
+
+func connectToMQTTBroker(loraURL string, logger logger.Logger) mqtt.Client {
+	opts := mqtt.NewClientOptions()
+	opts.AddBroker(loraURL)
+	opts.SetUsername("")
+	opts.SetPassword("")
+	opts.SetOnConnectHandler(func(c mqtt.Client) {
+		logger.Info("Connected to MQTT broker")
+	})
+	opts.SetConnectionLostHandler(func(c mqtt.Client, err error) {
+		logger.Error(fmt.Sprintf("MQTT connection lost: %s", err.Error()))
+		os.Exit(1)
+	})
+
+	client := mqtt.NewClient(opts)
+	if token := client.Connect(); token.Wait() && token.Error() != nil {
+		logger.Error(fmt.Sprintf("Failed to connect to Lora MQTT broker: %s", token.Error()))
+		os.Exit(1)
+	}
+
+	return client
+}
+
+func connectToRedis(redisURL, redisPass, redisDB string, logger logger.Logger) *r.Client {
+	db, err := strconv.Atoi(redisDB)
+	if err != nil {
+		logger.Error(fmt.Sprintf("Failed to connect to redis: %s", err))
+		os.Exit(1)
+	}
+
+	return r.NewClient(&r.Options{
+		Addr:     redisURL,
+		Password: redisPass,
+		DB:       db,
+	})
+}
+
+func connectToGRPC(url string, logger logger.Logger) *grpc.ClientConn {
+	conn, err := grpc.Dial(url,
+		grpc.WithTransportCredentials(credentials.NewClientTLSFromCert(nil, "")),
+		grpc.WithPerRPCCredentials(jwt{
+			token: "verysecret",
```

Comment:
Move string constants to the top of the file.

Reply:
I'll remove this function, we don't use it anymore